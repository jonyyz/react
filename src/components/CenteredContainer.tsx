import * as React from "react";

import {
  makeStyles,
  createStyles
} from "@material-ui/styles";

const useStyles = makeStyles(createStyles({
  container: {
    display: "flex",
    height: "100%",
    alignItems: "center",
    justifyContent: "center"
  },
}));

const CenteredContainer: React.FunctionComponent = ({ children }) => {
  const classes = useStyles({});

  return (
    <div className={classes.container}>{children}</div>
  );
}

export default CenteredContainer;
