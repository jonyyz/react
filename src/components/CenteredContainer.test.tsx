import * as React from "react";
import { shallow } from "enzyme";

import CenteredContainer from "./CenteredContainer";

describe("CenteredContainer", () => {
  test("should render with children", () => {
    const wrapper = shallow(
      <CenteredContainer>
        <div id="test">test</div>
      </CenteredContainer>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
