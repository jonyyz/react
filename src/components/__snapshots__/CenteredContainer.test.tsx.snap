// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredContainer should render with children 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    id="test"
  >
    test
  </div>
</div>
`;
